### PR TITLE
Minor code cleanup in Host

### PR DESCRIPTION
**Pull request by halfalicious**
Made some (very) minor changes while going through the Host code, should have no functional impact: 
* Add constexpr
* Add some minor comments
* Remove unnecessary ReputationManager constructor
* Use boost::asio namespace alias
* Update license messages

### CODE REVIEW

File: libp2p/Host.cpp [hunk 1]
```diff
@@ -1,18 +1,18 @@
 /*
-    This file is part of cpp-ethereum.
+    This file is part of aleth.
```
Comment (chfast) [inline libp2p/Host.cpp]: Please see #5403.

File: libp2p/Host.cpp [hunk 2]
```diff
@@ -99,7 +97,7 @@ Host::Host(string const& _clientVersion, KeyPair const& _alias, NetworkConfig co
     m_clientVersion(_clientVersion),
     m_netConfig(_n),
     m_ifAddresses(Network::getInterfaceAddresses()),
-    m_ioService(2),
+    m_ioService(2), // concurrency hint, suggests how many threads it should allow to run simultaneously
```
Comment (chfast) [inline libp2p/Host.cpp]: Use clang-format.

File: libp2p/Host.cpp [hunk 3]
```diff
@@ -705,7 +703,7 @@ void Host::run(boost::system::error_code const&)
     DEV_GUARDED(x_connecting)
         m_connecting.remove_if([](std::weak_ptr<RLPXHandshake> h){ return h.expired(); });
     DEV_GUARDED(x_timers)
-    m_timers.remove_if([](std::unique_ptr<boost::asio::deadline_timer> const& t) {
+    m_timers.remove_if([](std::unique_ptr<ba::deadline_timer> const& t) {
```
Comment (chfast) [inline libp2p/Host.cpp]: How about we use `using io = boost.asio`?

File: libp2p/Host.cpp [hunk 4]
```diff
@@ -757,6 +755,8 @@ void Host::run(boost::system::error_code const&)
     m_timer->async_wait(runcb);
 }
 
+// Called after thread has been started to perform additional class-specific state initialization (e.g. start
```
Comment (chfast) [inline libp2p/Host.cpp]: Maybe move the comment inside the function. This looks like doxygen comment.

File: libp2p/Host.cpp [hunk 5]
```diff
@@ -767,7 +767,7 @@ void Host::startedWorking()
         // time, stop will wait on m_timer and graceful network shutdown.
         Guard l(x_runTimer);
         // create deadline timer
-        m_timer.reset(new boost::asio::deadline_timer(m_ioService));
+        m_timer.reset(new ba::deadline_timer(m_ioService));
```
Comment (chfast) [inline libp2p/Host.cpp]: Do we need heap allocated timers? They cannot be used as values?

File: libp2p/Host.h [hunk 6]
```diff
@@ -87,7 +87,7 @@ struct Reputation
 class ReputationManager
 {
 public:
-    ReputationManager();
+    ReputationManager() = default;
```
Comment (chfast) [inline libp2p/Host.h]: Is this needed at all?

File: libp2p/Host.cpp (hunk 5 above)
Comment (gumb0) [reply libp2p/Host.cpp]: It looks we destroy and recreate it when the network is stopped / restarted, but I'm not sure if that's actually needed.

File: libp2p/Host.h (hunk 6 above)
Comment (halfalicious) [reply libp2p/Host.h]: Ah you're right, C++ default constructors are declared & defined implicitly so we don't need this.

File: libp2p/Host.cpp (hunk 4 above)
Comment (halfalicious) [reply libp2p/Host.cpp]: I like doxygen comments 😃 We have other functions that are commented like that e.g. https://github.com/ethereum/aleth/blob/b92056768376d42b809e2a80afd3178b2dead29f/libp2p/Common.cpp#L48-L51

Also, Geth functions are commented like that e.g: https://github.com/ethereum/go-ethereum/blob/472c23a8015cd84b193d2d0efb4592a664de3c62/internal/ethapi/addrlock.go#L43-L48

File: libp2p/Host.cpp (hunk 5 above)
Comment (halfalicious) [reply libp2p/Host.cpp]: I'd like to leave the timer alone for these changes (they were meant to be trivial changes with absolutely no potential for functional impact), I'd be happy to look at it later.

File: libp2p/Host.cpp [hunk 7]
```diff
@@ -759,6 +745,8 @@ void Host::run(boost::system::error_code const&)
 
 void Host::startedWorking()
 {
+    // Called after thread has been started to perform additional class-specific state initialization (e.g. start
+// capability threads, start TCP listener, and kick off timers)
```
Comment (chfast) [inline libp2p/Host.cpp]: Formatting off here

Comment (halfalicious) [reply libp2p/Host.cpp]: Thanks, will fix before merging

**Pull request merged**